### ISSUE TITLE
BPTC - CompressorSIMD.cpp doesn't compile

### ISSUE THREAD

**Issue by ivalylo** [GammaUNC/FasTC#20]
CompressorSIMD.cpp in BPTCEncoder doesn't compile. Multiple errors

**Comment by Mokosha** [comment]
Thanks for the feedback. We'll look into this. The current SIMD branch on master hasn't been updated in quite some time. We started [a separate ground-up rewrite](https://git.cs.unc.edu/git/?p=pavel/FasTC.git;a=shortlog;h=refs/heads/RedoSIMDPath) a few years ago, but that hasn't come close to being finished. In the meantime, if you'd like a fast, good SIMD compressor, I suggest you take a look at the [ISPC compressor](https://github.com/GameTechDev/ISPCTextureCompressor) from Intel.

**Comment by ivalylo** [comment]
It wasn't so hard to fix the SIMD files issues. I had a weird problem though. If I use the fast floating point mode in Visual Studio 2015, the generated texture is pixelated. If I use precise, it works as expected. I tried to change some fpu settings, but I'm not a SSE guru, so I didn't have any luck, will just live with the precise mode. 

I don't want to deal with the  ISPC compiler if possible. Also, I need something easy to hack that can give me reasonable result very fast. I was able to tweak FasTC to be a lot faster and still get decent result. It's nothing clever, just more error tolerance. But for my purpose, if I can get slightly worse result for 10x speed up, I would take it. On the Sponza scene for example, the NVIDIA driver can compress the 160 MB textures in 10 sec. The quality is pretty bad of course.  Currently, I'm getting 30 sec with FasTC on 6 core machine, which is not bad. My multi-threading is not optimal, since I load and compress in the same thread, so there is a lot of idling at the end. With better MT, it should be like 20 sec.

**Comment by Mokosha** [comment]
Hi, sorry I haven't responded in a while.

How did you compile it in the first place? I thought I turned off the SIMD codepath because it wasn't finished. Did you hack the CMake file?

**Comment by ivalylo** [comment]
No, I made a project in Visual Studio. Then tried to match the changes from the non-SIMD version. It wasn't much work, but of course, I'm not sure if everything was working.
